app/models/assignment.rb

```diff
@@ -268,10 +258,10 @@ def display_for_note
   end
 
   # Returns the maximum possible mark for a particular assignment
-  def max_mark(user_visibility = :ta)
+  def max_mark(user_visibility = :ta_visible)
     # TODO: sum method does not work with empty arrays. Consider updating/replacing gem:
     #       see: https://github.com/thirtysixthspan/descriptive_statistics/issues/44
-    max_marks = get_criteria(user_visibility).map(&:max_mark)
+    max_marks = criteria.where(user_visibility => true).map(&:max_mark)
```

Comment:
This method can be simplified now that it's a single query. Before, `get_criteria` returned an array, which why there is the `map` and the comment about summing an empty array.

But now you should be able to call `criteria.where(user_visibility => true).sum(:max_mark)` (or something similar) to grab the sum of the criteria max_marks, even when there are none.